src/wps/core/repository.py

```diff
@@ -207,7 +329,8 @@ async def get(
 
         In the following cases, a WorkPackageAccessError is raised:
         - if a work package with the given work_package_id does not exist
-        - if check_valid is set and the work package has expired
+        - if check_valid is set and the work package has expired or the box/dataset
+          ID is missing
```

Comment:
```suggestion
        - if check_valid is set and the work package has expired
```

the database should only contain work packages with existing dataset/box id